Dockerfile

```diff
@@ -397,7 +405,7 @@ RUN \
   echo "$X265_SHA256  x265.tar.bz2" | sha256sum --status -c - && \
   tar xf x265.tar.bz2 && \
   cd x265_*/build/linux && \
-  cmake -G "Unix Makefiles" -DENABLE_SHARED=OFF -DENABLE_AGGRESSIVE_CHECKS=ON -DCMAKE_ASM_NASM_FLAGS=-w-macro-params-legacy -DENABLE_CLI=OFF -DCMAKE_BUILD_TYPE=Release ../../source && \
+  cmake -G "Unix Makefiles" -DENABLE_SHARED=OFF -DENABLE_AGGRESSIVE_CHECKS=ON -DCMAKE_ASM_NASM_FLAGS=-w-macro-params-legacy -DENABLE_CLI=OFF -DENABLE_NASM=ON -DCMAKE_BUILD_TYPE=Release ../../source && \
```

Comment:
Was this off before? if so probably not good at all. But good to have it explicit i think.

Reply:
it is off by default

https://imgur.com/qqMpAMH.png

Reply:
Had a look at latest build log, it looks like it looks for nasm, ok good... i have a feeling x265 without assembly would be quite a lot slower. But i think explicit ON i good, hopefully it even fails if not found?
```
...
2022-02-17T20:02:27.4669997Z #17 2.208 CMake Warning (dev) at /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:438 (message):
2022-02-17T20:02:27.4670531Z #17 2.208   The package name passed to `find_package_handle_standard_args` (nasm) does
2022-02-17T20:02:27.4671010Z #17 2.208   not match the name of the calling package (Nasm).  This can lead to
2022-02-17T20:02:27.4671427Z #17 2.208   problems in calling code that expects `find_package` result variables
2022-02-17T20:02:27.4671827Z #17 2.208   (e.g., `_FOUND`) to follow a certain pattern.
2022-02-17T20:02:27.4672161Z #17 2.208 Call Stack (most recent call first):
2022-02-17T20:02:27.4672520Z #17 2.208   cmake/FindNasm.cmake:23 (find_package_handle_standard_args)
2022-02-17T20:02:27.4672893Z #17 2.208   CMakeLists.txt:352 (find_package)
2022-02-17T20:02:27.4673394Z #17 2.208 This warning is for project developers.  Use -Wno-dev to suppress it.
2022-02-17T20:02:27.4673729Z #17 2.208 
2022-02-17T20:02:27.4674118Z #17 2.209 -- Found nasm: /usr/bin/nasm (found version "2.15.05") 
2022-02-17T20:02:27.4674588Z #17 2.209 -- Found Nasm 2.15.05 to build assembly primitives
2022-02-17T20:02:27.4674996Z #17 2.222 -- GIT_EXECUTABLE /usr/bin/git
2022-02-17T20:02:27.4675428Z #17 2.222 -- SOURCE CODE IS FROM x265 GIT ARCHIVED ZIP OR TAR BALL
2022-02-17T20:02:27.5910999Z #17 2.224 -- GIT ARCHIVAL INFORMATION PROCESSED
2022-02-17T20:02:27.5911491Z #17 2.224 -- X265 RELEASE VERSION 3.5+1-f0c1022b6
2022-02-17T20:02:27.5911942Z #17 2.257 -- The ASM_NASM compiler identification is NASM
2022-02-17T20:02:27.5912357Z #17 2.259 -- Found assembler: /usr/bin/nasm
...
```

Reply:
`Nasm version 2.10.07 is too old. 2.13.0 or later required`
that would be my mistake i guess as i have ran the configure outside of a container on a centos7 server

Reply:
i believe that nasm check is just that, performs nothing more, according to cmake files i partyed with today. As you hinted at up the realm, it seems obvious that it would add ASM by default, being pretty much the universally expected bahavior when running configure on stuff that are bound to be enabled on a per system basis and package doesnt strictly requires their additiom/implementation. This was such a long phrase, felt like when my late Uncle Jiiimm recorded his country album, should compress and share, in Shiny format

I had been wanting to write this for hours now and figured i would shut about it but yet, i failed at keeping knowledge to myself - such a true canadian! (hahha no idea what that meant in context)

Can you see a place in the cmake files where it enables asm by default? No need to go all in on this, it might carve my insides so much i will have to verify and debug a full build

Reply:
> This guy's more PUNK than me!

Reply:
> i believe that nasm check is just that, performs nothing more, according to cmake files i partyed with today. As you hinted at up the realm, it seems obvious that it would add ASM by default, being pretty much the universally expected bahavior when running configure on stuff that are bound to be enabled on a per system basis and package doesnt strictly requires their additiom/implementation. This was such a long phrase, felt like when my late Uncle Jiiimm recorded his country album, should compress and share, in Shiny format

😄 

> I had been wanting to write this for hours now and figured i would shut about it but yet, i failed at keeping knowledge to myself - such a true canadian! (hahha no idea what that meant in context)

I'm swedish so maybe we're similar in this :)

> Can you see a place in the cmake files where it enables asm by default? No need to go all in on this, it might carve my insides so much i will have to verify and debug a full build

No haven't looked closer, a bit busy with other things, can have a look later if you don't beat me to it. If we really want to be sure about these things i guess we could add some kind of assert check looking at various program output (`x265 --version` etc) but maybe overkill. Would be nice to know if cmake generally fails if ones does `FEATURE=ON` and it can't enable FEATURE, if so maybe that is enough.

Reply:
no, it doesn't fail generally, check libssh build per example, it will let you know what it found that was out of place and what it 'did' to mitigate